tests/unit/model/relationships/has-many-test.js

```diff
@@ -906,6 +906,561 @@ test('it is possible to add a new item to a relationship', function(assert) {
+            name: 'Rambunctious'
+          }
+        }
+      ]
+    });
+  });
+
+  const person = store.peekRecord('person', '1');
+  let dog = run(() => person.get('dog'));
+  const shen = store.peekRecord('dog', '1');
+  const rambo = store.peekRecord('dog', '2');
+
+  assert.ok(dog === shen, 'precond - the belongsTo points to the correct dog');
+  assert.equal(get(dog, 'name'), 'Shenanigans', 'precond - relationships work');
+
+  run(() => {
+    person.set('dog', rambo);
+  });
+
+  dog = person.get('dog');
+  assert.equal(dog, rambo, 'precond2 - relationship was updated');
+
+  return run(() => {
+    return shen.destroyRecord({})
+      .then(() => {
+        shen.unloadRecord();
+
+        dog = person.get('dog');
+        assert.equal(dog, rambo, 'The currentState of the belongsTo was preserved after the delete');
+      });
+  });
+});
+
+test('new items added to an async belongsTo relationship are not cleared by a delete', function(assert) {
+  assert.expect(4);
+
+  const Person = DS.Model.extend({
+    name: DS.attr('string'),
+    dog: DS.belongsTo('dog', { async: true, inverse: null })
+  });
+
+  const Dog = DS.Model.extend({
+    name: DS.attr('string')
+  });
+
+  let env = setupStore({
+    person: Person,
+    dog: Dog
+  });
+  env.adapter.shouldBackgroundReloadRecord = () => false;
+  env.adapter.deleteRecord = () => {
+    return Ember.RSVP.Promise.resolve({ data: null });
+  };
+
+  let { store } = env;
+
+  run(() => {
+    store.push({
+      data: {
+        type: 'person',
+        id: '1',
+        attributes: {
+          name: 'Chris Thoburn'
+        },
+        relationships: {
+          dog: {
+            data: { type: 'dog', id: '1' }
+          }
+        }
+      },
+      included: [
+        {
+          type: 'dog',
+          id: '1',
+          attributes: {
+            name: 'Shenanigans'
+          }
+        },
+        {
+          type: 'dog',
+          id: '2',
+          attributes: {
+            name: 'Rambunctious'
+          }
+        }
+      ]
+    });
+  });
+
+  return run(() => {
+    const person = store.peekRecord('person', '1');
+    const shen = store.peekRecord('dog', '1');
+    const rambo = store.peekRecord('dog', '2');
+
+    return person.get('dog').then((dog) => {
+      assert.ok(dog === shen, 'precond - the belongsTo points to the correct dog');
+      assert.equal(get(dog, 'name'), 'Shenanigans', 'precond - relationships work');
+
+      person.set('dog', rambo);
+
+      dog = person.get('dog.content');
+
+      assert.ok(dog === rambo, 'precond2 - relationship was updated');
+
+      return shen.destroyRecord({})
+        .then(() => {
+          shen.unloadRecord();
+
+          dog = person.get('dog.content');
+          assert.ok(dog === rambo, 'The currentState of the belongsTo was preserved after the delete');
+        });
+    });
+  });
+});
+
+test('deleting an item that is the current state of a belongsTo restores canonicalState', function(assert) {
```

Comment:
why?

1. this is a change in semantics (see [this twiddle](https://ember-twiddle.com/7f94a0e81ea38d9022a5007ef8be15c8?openFiles=adapters.application.js%2C))
2. this is surprising behaviour